contracts/factories/BridgeFactory.sol

```diff
@@ -0,0 +1,274 @@
+      _arcadeumCoinAddr != address(0) && 
+      _periodMintLimit > 0,
+      "BridgeFactory#constructor: INVALID_INPUT"
+    );
+
+    // Assets
+    skyweaverAssets = ISkyweaverAssets(_assetsAddr);
+    arcadeumCoin = IERC1155(_arcadeumCoinAddr);
+    arcadeumCoinID = _arcadeumCoinID;
+
+    // Set current period
+    period = livePeriod();
+    availableSupply = _periodMintLimit;
+
+    // Rewards parameters
+    periodMintLimit = _periodMintLimit;
+    emit PeriodMintLimitChanged(0, _periodMintLimit);
+  }
+
+
+  /***********************************|
+  |         Management Methods        |
+  |__________________________________*/
+
+  /**
+   * @notice Will update the daily mint limit
+   * @dev This change will take effect immediatly once executed
+   * @param _newPeriodMintLimit Amount of assets that can be minted within 24h
+   */
+  function updatePeriodMintLimit(uint256 _newPeriodMintLimit) external onlyOwnerTier(HIGHEST_OWNER_TIER) {
+    // Immediately update supply instead of waiting for next period
+    if (availableSupply > _newPeriodMintLimit) {
+      availableSupply = _newPeriodMintLimit;
+    }
+
+    emit PeriodMintLimitChanged(periodMintLimit, _newPeriodMintLimit);
+    periodMintLimit = _newPeriodMintLimit;
+  }
+
+  /**
+   * @notice Send current ARC balance of conquest contract to recipient
+   * @param _recipient Address where the currency will be sent to
+   * @param _data      Data to pass with transfer function
+   */
+  function withdraw(address _recipient, bytes calldata _data) external onlyOwnerTier(HIGHEST_OWNER_TIER) {
+    require(_recipient != address(0x0), "BridgeFactory#withdraw: INVALID_RECIPIENT");
+    uint256 thisBalance = arcadeumCoin.balanceOf(address(this), arcadeumCoinID);
+    arcadeumCoin.safeTransferFrom(address(this), _recipient, arcadeumCoinID, thisBalance, _data);
+  }
+
+
+  /***********************************|
+  |      Receiver Method Handler      |
+  |__________________________________*/
+
+  /**
+   * @notice Prevents receiving Ether or calls to unsuported methods
+   */
+  fallback () external {
+    revert("BridgeFactory#_: UNSUPPORTED_METHOD");
+  }
+
+  /**
+   * @notice Burns SW asset when received, or store ARC
+   * @dev Make sure to send the right assets and the correct amount 
+   *      as it's not validated in this contract.
+   * @param _id      Id of Token being transferred
+   * @param _amount  Amount of Token _id being transferred
+   */
+  function onERC1155Received(
+    address, // _operator
+    address, // _from 
+    uint256 _id, 
+    uint256 _amount, 
+    bytes memory // _data
+  )
+    public override returns(bytes4)
+  {
+    if (msg.sender == address(arcadeumCoin)) {
+      // Do nothing, just store the ARC until withdrawl
+      require(_id == arcadeumCoinID, "BridgeFactory#onERC1155Received: INVALID_ARC_ID");
+
+    } else if (msg.sender == address(skyweaverAssets)) {
+      // Burn asset received
+      skyweaverAssets.burn(_id, _amount);
```

Comment:
Use `msg.sender` instead of `skyweaverAssets` to avoid SLOADing twice

Reply:
That's a good idea :)